File: AFView/Pod/Views/AFActionSheet.m
#import "AFActionSheet.h"


#pragma mark Class Definition

@implementation AFActionSheet
{
	@private __strong AFActionSheetCompletion _completion;
	@private __strong NSArray *_options;
	
	@private NSInteger _cancelButtonIndex;
}


#pragma mark - Public Methods

+ (void)presentFromViewController: (UIViewController *)viewController
	title: (NSString *)title
	otherButtonTitles: (NSArray *)otherButtonTitles
	completion: (AFActionSheetCompletion)completion
{
	// Keep around the completion callback.
	__block AFActionSheetCompletion blockCompletion = [completion copy];

	UIAlertController *actionSheet = [UIAlertController alertControllerWithTitle: title
		message: nil
		preferredStyle: UIAlertControllerStyleActionSheet];

	// TODO: LOCALIZE:
    [actionSheet addAction: [UIAlertAction actionWithTitle: @"Cancel"
    	style: UIAlertActionStyleCancel
		handler: ^(UIAlertAction *action) {
			
			// Call back.
        	[viewController dismissViewControllerAnimated: YES completion: ^{
        		if (blockCompletion != nil)
				{
					blockCompletion(nil);
				}
			}];
    }]];

	// Add each button title.
	for (NSString *buttonTitle in otherButtonTitles)
	{
		[actionSheet addAction: [UIAlertAction actionWithTitle: buttonTitle
			style: UIAlertActionStyleDefault
			handler: ^(UIAlertAction *action) {
				
				// Call back.
	    		[blockCompletion dismissViewControllerAnimated: YES
					completion: ^{
					if (blockCompletion != nil)
					{
						blockCompletion(buttonTitle);
					}
        		}];
    	}]];
	}

    // Present action sheet.
    [viewController presentViewController: actionSheet
		animated: YES
		completion: nil];
}


@end

File: AFView/Pod/Views/AFActionSheet.swift
import Foundation

public typealias AFActionSheetCompletion = (selection: String?) -> Void

public class AFActionSheet : NSObject
{
	// MARK: - Properties
	
	private var _completion: AFActionSheetCompletion? = nil
	private var _options: [String]? = nil
	
	
	// MARK: - Public Methods
	
	public func presentFromViewController(viewController: UIViewController,
		title: String,
		otherButtonTitles: [String],
		completion: AFActionSheetCompletion)
	{
		// Only allow a single call.
		if _completion != nil
		{
			if let completion: AFActionSheetCompletion = completion
			{
				completion(selection: nil)
			}
			return
		}
	
		// Track completion.
		_completion = completion
		
		let alertController: UIAlertController = UIAlertController(title: title,
			message: nil,
			preferredStyle: .ActionSheet)
		
		// Add the cancel button.
		let cancelAction: UIAlertAction = UIAlertAction(title: "Cancel",
			style: .Default,
			handler:
			{ (action) in
			
				// Completion with a nil result.
				if let _completion = self._completion
				{
					_completion(selection: nil)
				}
			})
		
		// Add the action.
		alertController.addAction(cancelAction)
		
		// Add each button title.
		for buttonTitle in otherButtonTitles
		{
			// Create actions for each other buttons.
			let action: UIAlertAction = UIAlertAction(title: buttonTitle,
				style: .Default,
				handler:
				{ (action) in
				
					// Completion with the selected result.
					if let _completion = self._completion
					{
						_completion(selection: buttonTitle)
					}
				})
		
			// Add the action.
			alertController.addAction(action)
		}
		
		// Present the sheet.
		viewController.presentViewController(alertController,
			animated: true,
			completion: nil)
	}
}